src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -491,12 +493,23 @@ export class StudyViewPageStore {
     });
 
     private readonly samples = remoteData<Sample[]>({
+        await: () => [this.clinicalAttributes],
         invoke: () => {
             return internalClient.fetchFilteredSamplesUsingPOST({
                 studyViewFilter: this.emptyFilter
             })
         },
-        default: []
+        default: [],
+        onResult: (result) => {
+            this._sampleClinicalDataSet = _.reduce(result, (acc, next) => {
```

Comment:
seems that this is a computation of samples, so why not just make it a remoteData that awaits samples and then spits back the transformed value?  or even just a helper that consuming code can use to transform the data?   maybe there's a reason here, but we want to avoid using onResult to set data except when absolutely necessary.  it's an antipattern.